src/main/java/mate/academy/springboot/datajpa/controller/ProductController.java

```diff
@@ -0,0 +1,82 @@
+package mate.academy.springboot.datajpa.controller;
+
+import java.math.BigDecimal;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import mate.academy.springboot.datajpa.dto.ProductRequestDto;
+import mate.academy.springboot.datajpa.dto.ProductResponseDto;
+import mate.academy.springboot.datajpa.dto.mapper.ProductMapper;
+import mate.academy.springboot.datajpa.model.Category;
+import mate.academy.springboot.datajpa.model.Product;
+import mate.academy.springboot.datajpa.service.CategoryService;
+import mate.academy.springboot.datajpa.service.ProductService;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/products")
+public class ProductController {
+    private final ProductService productService;
+    private final ProductMapper productMapper;
+    private final CategoryService categoryService;
+
+    public ProductController(ProductService productService, ProductMapper productMapper,
+                             CategoryService categoryService) {
+        this.productService = productService;
+        this.productMapper = productMapper;
+        this.categoryService = categoryService;
+    }
+
+    @PostMapping
+    public ProductResponseDto create(@RequestBody ProductRequestDto requestDto) {
+        Product product = productMapper.toModel(requestDto);
+        productService.save(product);
+        return productMapper.toResponseDto(product);
+    }
+
+    @PutMapping("/{id}")
+    public ProductResponseDto update(@PathVariable Long id,
+                                      @RequestBody ProductRequestDto requestDto) {
+        Product product = productMapper.toModel(requestDto);
+        product.setId(id);
+        productService.update(product);
+        return productMapper.toResponseDto(product);
+    }
+
+    @DeleteMapping("/{id}")
+    public void delete(@PathVariable Long id) {
+        productService.delete(id);
+    }
+
+    @GetMapping("/{id}")
+    public ProductResponseDto getById(@PathVariable Long id) {
+        Product product = productService.get(id);
+        return productMapper.toResponseDto(product);
+    }
+
+    @GetMapping("/by-price")
```

Comment:
In my mind it's better to use only "price", and this would looking like "products/price?from=1&&to=2"
Like there 
<img width="143" alt="image" src="https://user-images.githubusercontent.com/95774539/209564342-7b3eaf8a-5cf8-4720-a7d8-a3ce8d236516.png">